### PR TITLE
#192 Integrated emails and triggers for components, bikes and roles

### LINKED ISSUE #192: Integrate triggers and emails

### PR DESCRIPTION
**Pull request by daniela-venuta**
Resolves #192

### CODE REVIEW

File: bike-erp/backend/src/main/routes/component_routes.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ router.get("/", authenticateToken, (req, res) => {
 });
 
 //Backend endpoint to get all the component types in categories
-router.get("/componentTypes", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE]), (req, res) => {
+router.get("/componentTypes", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE, Role.CUSTOMER]), (req, res) => {
```
Comment (Manraj45) [inline bike-erp/backend/src/main/routes/component_routes.ts:24]: If all roles are allowed than you can remove the verify role method

Comment (daniela-venuta) [reply bike-erp/backend/src/main/routes/component_routes.ts:24]: @nasaku898 this was done to fix a bug, should we keep it like this so we know what is affected or do I remove the verify?

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Everything seems good, a demo would be appreciated. For the roles restriction on endpoints, If all roles are allowed than you can remove the verify role method from the endpoint.

**Review by Mnx458 — APPROVED** [review]

**Review by bowsong — APPROVED** [review]
look good

**Review by Manraj45 — APPROVED** [review]
Good Job!

**Pull request merged**